### PR TITLE
docs: #1231 adding retention.local.target to Manage Disk Space page

### LINKED ISSUE #1231: advice for setting retention.local.target.*
### Describe the Issue

check for latest posts: https://redpandadata.slack.com/archives/C02FSLH3ZS9/p1676352207070739
summary
Product feedback from Overhaul:
They've hit an issue where nodes in the cluster were about to run out of the disk space, which led to all writes being blocked. The issue itself is mitigated, but  they're expecting redpanda to be more intelligent as to local disk management; e.g. 1) dynamically tweaking the eviction rate in local storage based on the workload and free disk 2) recalculate the retention_local_target_bytes for every topic as a % of disk space every time a topic or partitions were added. Is this something we're thinking about or otherwise?

Alexander Gallego (PDT; San Francisco)
i also think is a gap in docs 
I think retention.local.target.bytes should be used under specific conditions and not set by the majority of users.

Matt Schumpert
 at a minimum we'll want to document that cloud explicitly sets it to 12 hours.

Matt Schumpert
In platform docs, Tiered Storage doc describes it well: https://docs.redpanda.com/docs/manage/tiered-storage/#manage-local-capacity-for-tiered-storage-topics
However,  we need a mention here too (local target is absent there): https://docs.redpanda.com/docs/manage/cluster-maintenance/disk-utilization/#configure-message-retention docs.redpanda.com

Michele Cyran
 Could we just add a link in [Configure message retention](https://docs.redpanda.com/docs/manage/cluster-maintenance/disk-utilization/#configure-message-retention)?
For details about configuring local retention in a Tiered Storage implementation, see [Manage local capacity for Tiered Storage topics](https://docs.redpanda.com/docs/manage/tiered-storage/#manage-local-capacity-for-tiered-storage-topics).

Matt Schumpert
 I think it's important to have something in the 'manage disk space'.  A customer ran out of disk space because they didn't understand this.  If I were an admin setting up a self-hosted cluster I'd read the 'manage disk space' part first.  We could link to the details in the Tiered Storage / Message Retention pages from there, but this is a capacity planning thing people need to know up front (edited)

SME = Piyush, Matt

### Updates to existing documentation

 [Configure message retention](https://docs.redpanda.com/docs/manage/cluster-maintenance/disk-utilization/#configure-message-retention)

### PR DESCRIPTION
**Pull request by micheleRP**
resolves #1231 

This adds info about `retention.local.target` to the Manage Disk Space page (based on [this slack post](https://redpandadata.slack.com/archives/C02FSLH3ZS9/p1676352207070739)). 

preview: https://deploy-preview-1279--redpanda-documentation.netlify.app/docs/manage/cluster-maintenance/disk-utilization/#local-retention-with-tiered-storage

### CODE REVIEW

File: versioned_docs/version-22.3/manage/cluster-maintenance/disk-utilization.mdx [hunk 1]
```diff
@@ -90,6 +90,16 @@ To set retention size for a single topic, use `retention.bytes`, which overrides
   rpk cluster config set retention_bytes <retention_size>
   ```
 
+### Local retention with Tiered Storage
+
+Redpanda Tiered Storage provides the ability to offload log segments to cloud storage in near real time. Tiered Storage can be combined with local storage to provide long-term data retention and disaster recovery on a per-topic basis. 
+
+You can use topic-level properties to control retention of topic data in local storage. Similar to `retention.ms` and `retention.bytes`, data expires from local storage following `retention.local.target.ms` or `retention.local.target.bytes`. The properties default to an optimized local data expiration. 
+
+When set, Redpanda keeps actively-used and sequential (next segment) data in the local cache and targets to maintain this age of data in local storage. It purges data based on actual available local volume space, without forcing disk full situations when there is data skew. Redpanda Cloud sets `retention.local.target` to 12 hours.
```
Comment (jcsp) [inline versioned_docs/version-22.3/manage/cluster-maintenance/disk-utilization.mdx:99]: The guidance about default local retention using local disk space should be the first sentence in the section -- that's the motivation for having text here as well as in the "manage local capacity..." page.

Something like:

> Topics using Redpanda Tiered Storage will keep the most recent 24 hours of data on local disk by default (12 hours in Redpanda Cloud).  It is possible to exhaust local disk even when using tiered storage, if data is written fast enough to consume all disk space within the local retention time period: this period can be adjusted if required.

... then a second paragraph to say what the settings are and link to the other page for more detail.

File: versioned_docs/version-22.3/manage/cluster-maintenance/disk-utilization.mdx [hunk 2]
```diff
@@ -114,7 +124,7 @@ Compaction saves space by retaining only the last known value for each message k
 
 When compaction is configured, topics take their size from `compacted_log_segment_size`. The `log_segment_size` property does not apply to compacted topics. 
 
-Setting a `segment.bytes` size on a topic applies whether the topic is compacted or not, and the `max_compacted_log_segment_size` property applies to compacted topics regardless of the other properties. It controls how many segments are merged together. For example, if you set `segment.bytes` to 128MB, but leave `max_compacted_log_segment_size` at 5GB, you get 128MB segments when they're written, but up to 5GB segments after compaction.
+Setting a `segment.bytes` size on a topic applies whether the topic is compacted or not, and the `max_compacted_log_segment_size` property applies to compacted topics regardless of the other properties. It controls how many segments are merged together. For example, if you set `segment.bytes` to 128 MB, but leave `max_compacted_log_segment_size` at 5 GB, you get 128 MB segments when they're written, but up to 5 GB segments after compaction.
```
Comment (Feediver1) [inline versioned_docs/version-22.3/manage/cluster-maintenance/disk-utilization.mdx:127]:
```suggestion
Setting a `segment.bytes` size on a topic applies whether the topic is compacted or not, and the `max_compacted_log_segment_size` property applies to compacted topics regardless of any other properties. The `max_compacted_log_segment_size` property controls how many segments are merged together. For example, if you set `segment.bytes` to 128 MB, but leave `max_compacted_log_segment_size` at 5 GB, then you get 128 MB segments when they're written, but up to 5 GB segments after compaction.
```

**Review by Feediver1 — APPROVED** [review]
Left some suggestions.

File: versioned_docs/version-22.3/manage/cluster-maintenance/disk-utilization.mdx (hunk 1 above)
Comment (micheleRP) [reply versioned_docs/version-22.3/manage/cluster-maintenance/disk-utilization.mdx]: Thanks @jcsp and @Feediver1! Please review this updated section: https://deploy-preview-1279--redpanda-documentation.netlify.app/docs/manage/cluster-maintenance/disk-utilization/#configure-message-retention. 

@shorgi: It looks like `retention_bytes` is included [here](https://docs.redpanda.com/docs/reference/cluster-properties/#retention_bytes), but not `retention.ms`, `retention.local.target.ms`, or `retention.local.target.bytes`.

File: versioned_docs/version-22.3/manage/cluster-maintenance/disk-utilization.mdx [hunk 3]
```diff
@@ -19,9 +19,9 @@ When a Redpanda node runs out of disk space, it terminates. This causes leader r
...
+By default, all self-hosted topics using Redpanda Tiered Storage keep the last 24 hours of data on local disk. Redpanda Cloud keeps the last 12 hours of data. However, if data is written fast enough to consume all disk space within the local retention time period, it's possible to exhaust local disk space, even when using Tiered Storage. You can adjust the default local storage size or time period, if required.
 
-You can configure retention properties to delete messages based on the following conditions:
+Retention properties control how long messages are kept on disk before they're deleted or compacted. Setting message retention properties is the best way to prevent old messages from accumulating on disk to the point that the disk becomes full. When set, Redpanda keeps actively-used and sequential (next segment) data in the local cache and targets to maintain this age of data in local storage. It purges data based on actual available local volume space, without forcing disk full situations when there is data skew. You can configure retention properties to delete messages based on the following conditions:
```
Comment (jcsp) [inline versioned_docs/version-22.3/manage/cluster-maintenance/disk-utilization.mdx:24]: >When set, Redpanda keeps actively-used and sequential (next segment) data in the local cache and targets to maintain this age of data in local storage. It purges data based on actual available local volume space, without forcing disk full situations when there is data skew.

These sentences are a bit ambiguous and could cause confusion between the local raft storage (governed by retention settings) and the local tiered storage cache (governed by a fixed cache size).  I think we can perhaps just remove these sentences and stick to explaining the retention settings.

File: versioned_docs/version-22.3/manage/cluster-maintenance/disk-utilization.mdx [hunk 4]
```diff
@@ -31,12 +31,17 @@ Set retention properties at the topic level or the cluster level. If a value isn
 
 | Retention property | Cluster level | Topic level <br /> (overrides cluster configuration) |
 | ---               | ---                                                |  ---                                 |
-| Time-based        | `delete_retention_ms` <br /> Default - `604800000` | `retention.ms` <br /> No default     |
-| Size-based        | `retention_bytes`  <br /> Default - `null`         | `retention.bytes`  <br /> No default |
-| Segment size      | `log_segment_size`  <br /> Default - `1073741824`  | N/A                                  |
+| Time-based        | `delete_retention_ms` <br /> Default - `604800000` | `retention.ms`, `retention.local.target.ms` <br /> No default     |
```
Comment (jcsp) [inline versioned_docs/version-22.3/manage/cluster-maintenance/disk-utilization.mdx:34]: Can we add the cluster level `retention_local_target_ms_default` here?  And `retention_local_target_bytes_default` for the next row.

File: versioned_docs/version-22.3/manage/cluster-maintenance/disk-utilization.mdx [hunk 5]
```diff
@@ -19,9 +19,9 @@ When a Redpanda node runs out of disk space, it terminates. This causes leader r
 
 ## Configure message retention
```
[the remark this answers is not in the document]
Comment (micheleRP) [reply versioned_docs/version-22.3/manage/cluster-maintenance/disk-utilization.mdx:20]: Please see https://github.com/redpanda-data/documentation/issues/1042

File: versioned_docs/version-22.3/manage/cluster-maintenance/disk-utilization.mdx (hunk 4 above)
Comment (micheleRP) [reply versioned_docs/version-22.3/manage/cluster-maintenance/disk-utilization.mdx]: Thanks, @jcsp! Please review the [updated table](https://deploy-preview-1279--redpanda-documentation.netlify.app/docs/manage/cluster-maintenance/disk-utilization/#configure-message-retention).  Should the descriptions be different between the Tiered-storage vs non-Tiered properties? Something like Time-based (Tiered storage only), where Tiered Storage is where the *local*  properties take effect?

Comment (jcsp) [reply versioned_docs/version-22.3/manage/cluster-maintenance/disk-utilization.mdx]: Perhaps, I'm not sure of the best way to present it.  It seems like the prose is clear on the way the local.target settings control local retention in the context of tiered storage, whereas other settings control the overall retention, but maybe there's a way to make that clearer in the table as well.

**Review by jcsp — APPROVED** [review]

**Comment by mattschumpert** [comment]
As a convenience, we could convert to useful units like hours, e.g. default: '86400000' (24 hours).  Besides that looks good

**Comment by micheleRP** [comment]
Soon the properties will become links into the source reference page, so we won't show the defaults here. After Edward adds the new topic properties page, I'll change all properties in that table to links; e.g., https://docs.redpanda.com/docs/reference/cluster-properties/#delete_retention_ms

**Pull request merged**